develop/dev-guide-sample-application-python-mysqlclient.md

```diff
@@ -1,292 +1,238 @@
+- Prepare your environment. 
+- Connect to TiDB Serverless, TiDB Dedicated or TiDB Self-hosted using mysqlclient.
+- Build and run your app. Optionally, you can find sample code snippets for basic CRUD operations. 
 
-**Use a TiDB Serverless cluster**
-
-For detailed steps, see [Create a TiDB Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md#step-1-create-a-tidb-serverless-cluster).
-
-**Use a local cluster**
+> **Note:**
+> This tutorial works with TiDB Serverless, TiDB Dedicated and TiDB Self-hosted.
 
-For detailed steps, see [Deploy a local test cluster](/quick-start-with-tidb.md#deploy-a-local-test-cluster) or [Deploy a TiDB cluster using TiUP](/production-deployment-using-tiup.md).
+## Prerequisites
 
-</CustomContent>
+- [Python **3.10** or higher](https://www.python.org/downloads/).
+- [Git](https://git-scm.com/downloads).
+- A TiDB cluster. If you don't have a TiDB cluster, you can create one as follows:
+    - (Recommended) Follow [Creating a TiDB Serverless Cluster](/dev-guide-build-cluster-in-cloud.md) to create your own TiDB Cloud cluster.
+    - Follow [Deploy a Local Test TiDB Cluster](../quick-start-with-tidb.md#deploy-a-local-test-cluster) or [Deploy a Production TiDB Cluster](../production-deployment-using-tiup.md) to create a local cluster.
 
-<CustomContent platform="tidb-cloud">
+## Run sample app to connect to TiDB
 
-See [Create a TiDB Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md#step-1-create-a-tidb-serverless-cluster).
+This section demonstrates how to run the sample application code and connect to TiDB.
 
-</CustomContent>
+### Step 1: Clone the sample app repository
 
-## Step 2. Get the code
+Run the following command in your terminal window to clone the sample code repository.
 
-```shell
+```bash
 git clone https://github.com/pingcap-inc/tidb-example-python.git
 ```
 
-The following uses mysqlclient 2.1.1 as an example. Drivers for Python are more convenient to use than other languages, but they do not shield the underlying implementation and require manual management of transactions. If there are not a lot of scenarios where SQL is required, it is recommended to use ORM, which can help reduce the coupling of your program.
+### Step 2: Install dependencies (including mysqlclient)
 
-```python
-import uuid
-from typing import List
-
-import MySQLdb
-from MySQLdb import Connection
-from MySQLdb.cursors import Cursor
-
-def get_connection(autocommit: bool = True) -> MySQLdb.Connection:
-    return MySQLdb.connect(
-        host="127.0.0.1",
-        port=4000,
-        user="root",
-        password="",
-        database="test",
-        autocommit=autocommit
-    )
+Run the following commands to install required packages for the sample app.
 
+```bash
+cd tidb-example-python;
+pip install -r requirement.txt
+```
 
-def create_player(cursor: Cursor, player: tuple) -> None:
-    cursor.execute("INSERT INTO player (id, coins, goods) VALUES (%s, %s, %s)", player)
```

Comment:
Maybe using table name `players` is better than `player`:

- using `accounts` as table name in CockroachDB's sample code
- using `users` as table name in Planetscale's sample code
- using `movies` / `countries` as table names in Supabase's sample code

However, there are also examples of using singular nouns as table name:

- using `employee` as table name in the YugabyteDB sample code